views_api.py

```diff
@@ -84,13 +83,28 @@ async def api_add_track(
     else:
         raise TypeError("need either producer_id or producer_name arguments")
 
-    if id:
-        await update_track(
-            ls.id, id, data.name, data.download_url, data.price_msat or 0, p_id
-        )
+    return await add_track(
+        ls.id, data.name, data.download_url, data.price_msat or 0, p_id
+    )
+
+
+@livestream_ext.put("/api/v1/livestream/tracks/{id}")
```

Comment:
- a good bunch of duplicate code here
- better make `add_track` an [UPSERT](https://www.postgresqltutorial.com/postgresql-tutorial/postgresql-upsert/) (update on conflict)